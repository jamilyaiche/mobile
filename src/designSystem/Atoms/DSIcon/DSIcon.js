import PropTypes from 'prop-types'
import React from 'react'
import { TouchableOpacity } from 'react-native'
import { createIconSetFromFontello } from 'react-native-vector-icons'
import fontelloConfig from '../../../assets/fontIconsConfig/config.json'
import Colors from '../../../assets/styles/Colors'

const Icon = createIconSetFromFontello(fontelloConfig)

const DSIcon = (props) => {
  let size = 20
  if (props.size === 'xxx-small') {
    size = 5
  }
  if (props.size === 'extra-small') {
    size = 14
  }
  if (props.size === 'medium') {
    size = 40
  }
  if (props.size === 'large') {
    size = 60
  }
  if (props.size === 'extra-large') {
    size = 80
  }
  return (
    <TouchableOpacity
      {...(props.onLongPress !== undefined && { onLongPress: props.onLongPress })}
      {...(props.onPress !== undefined && { onPress: props.onPress })}
      {...(props.onPressIn !== undefined && { onPressIn: props.onPressIn })}
      {...(props.onPressOut !== undefined && { onPressOut: props.onPressOut })}
      {...(props.style !== undefined && { style: [props.style, { padding: 1 }] })}
      {...(props.disabled !== undefined && { disabled:props.disabled})}
    >
      <Icon name={props.name} size={size} color={props.color} />
    </TouchableOpacity>
  )
}

DSIcon.defaultProps = {
  color: Colors.black,
  size: 'small', // 'small', 'medium' , 'large', 'extra-large',
  name: 'eye'
}

DSIcon.propTypes = {
  onLongPress: PropTypes.func,
  onPress: PropTypes.func,
  onPressIn: PropTypes.func,
  onPressOut: PropTypes.func,
  name: PropTypes.string,
  size: PropTypes.oneOf(['xxx-small', 'extra-small', 'small', 'medium', 'large', 'extra-large']),
  color: PropTypes.string
}

export default DSIcon
