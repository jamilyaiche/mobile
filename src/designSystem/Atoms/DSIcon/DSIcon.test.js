import React from 'react';
import renderer from 'react-test-renderer'
import DSIcon from "../../Atoms/DSIcon/DSIcon"
import { createIconSetFromIcoMoon } from 'react-native-vector-icons'
import fontelloConfig from '../../../assets/fontIconsConfig/selection.json'
import {TouchableOpacity} from "react-native"

const Icon = createIconSetFromIcoMoon(fontelloConfig)

describe('DSIcon tests',()=>{
    it('test DSIcon snapshot',()=>{
        const wrapper=renderer.create(<DSIcon name="camera" size="medium" color="red" />)
        .toJSON()
        expect(wrapper).toMatchSnapshot()
    })

    it('test DSIcon size small=20',()=>{
        const wrapper=shallow(<DSIcon name="camera" size="small" color="red" />)
        expect(wrapper.find(TouchableOpacity).first().children().first().props().size===20)
        .toEqual(true)
    })

    it('test DSIcon size medium=40',()=>{
        const wrapper=shallow(<DSIcon name="camera" size="medium" color="red" />)
        expect(wrapper.find(TouchableOpacity).first().children().first().props().size===40)
        .toEqual(true)
    })

    it('test DSIcon size large=60',()=>{
        const wrapper=shallow(<DSIcon name="camera" size="large" color="red" />)
        expect(wrapper.find(TouchableOpacity).first().children().first().props().size===60)
        .toEqual(true)
    })

    it('test DSIcon size extra-large=20',()=>{
        const wrapper=shallow(<DSIcon name="camera" size="extra-large" color="red" />)
        expect(wrapper.find(TouchableOpacity).first().children().first().props().size===80)
        .toEqual(true)
    })

    it('test DSIcon onPress',()=>{
        const press1=jest.fn()
        const wrapper=renderer.create(<DSIcon name="camera" size="small" color="red" onPress={press1} />)
        wrapper.root.props.onPress()
        expect(press1.mock.calls.length).toEqual(1)
    })

    it('test DSIcon onLongPress',()=>{
        const press1=jest.fn()
        const wrapper=renderer.create(<DSIcon name="camera" size="small" color="red" onLongPress={press1} />)
        wrapper.root.props.onLongPress()
        expect(press1.mock.calls.length).toEqual(1)
    })

    it('test DSIcon onPressIn',()=>{
        const press1=jest.fn()
        const wrapper=renderer.create(<DSIcon name="camera" size="small" color="red" onPressIn={press1} />)
        wrapper.root.props.onPressIn()
        expect(press1.mock.calls.length).toEqual(1)
    })

    it('test DSIcon onPressOut',()=>{
        const press1=jest.fn()
        const wrapper=renderer.create(<DSIcon name="camera" size="small" color="red" onPressOut={press1} />)
        wrapper.root.props.onPressOut()
        expect(press1.mock.calls.length).toEqual(1)
    })
})