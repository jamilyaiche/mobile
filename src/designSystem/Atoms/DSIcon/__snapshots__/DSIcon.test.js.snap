// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DSIcon tests test DSIcon snapshot 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "red",
          "fontSize": 40,
        },
        undefined,
        Object {
          "fontFamily": "fontello",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    Ó†è
  </Text>
</View>
`;
